src/lib/ConfettiExplosion.svelte

```diff
@@ -310,68 +276,54 @@
+
+		// Get x landing point for it
+		setCSSVar(
+			'--x-landing-point',
+			`${mapRange(abs(rotate(degree, 90) - 180), 0, 180, -stageWidth / 2, stageWidth / 2)}px`
+		);
+		setCSSVar('--duration-chaos', `${duration - mathRound(random() * 1e3)}ms`);
 
 		// x-axis disturbance, roughly the distance the particle will initially deviate from its target
-		const x1 = shouldBeCrazy ? round(Math.random() * CRAZY_PARTICLE_CRAZINESS, 2) : 0;
-		const x2 = x1 * -1;
-		const x3 = x1;
+		const x1 =
+			random() < CRAZY_PARTICLES_FREQUENCY ? round(random() * CRAZY_PARTICLE_CRAZINESS, 2) : 0;
+		setCSSVar('--x1', `${x1}`);
+		setCSSVar('--x2', `${x1 * -1}`);
+		setCSSVar('--x3', `${x1}`);
 		// x-axis arc of explosion, so 90deg and 270deg particles have curve of 1, 0deg and 180deg have 0
-		const x4 = round(Math.abs(mapRange(Math.abs(rotate(degree, 90) - 180), 0, 180, -1, 1)), 4);
+		setCSSVar('--x4', `${round(abs(mapRange(abs(rotate(degree, 90) - 180), 0, 180, -1, 1)), 4)}`);
 
 		// roughly how fast particle reaches end of its explosion curve
-		const y1 = round(Math.random() * BEZIER_MEDIAN, 4);
+		setCSSVar('--y1', `${round(random() * BEZIER_MEDIAN, 4)}`);
 		// roughly maps to the distance particle goes before reaching free-fall
-		const y2 = round(Math.random() * force * (coinFlip() ? 1 : -1), 4);
+		setCSSVar('--y2', `${round(random() * force * (coinFlip() ? 1 : -1), 4)}`);
 		// roughly how soon the particle transitions from explosion to free-fall
-		const y3 = BEZIER_MEDIAN;
+		setCSSVar('--y3', `${BEZIER_MEDIAN}`);
 		// roughly the ease of free-fall
-		const y4 = round(Math.max(mapRange(Math.abs(degree - 180), 0, 180, force, -force), 0), 4);
-
-		const setCSSVar = (key: string, val: string | number) => node.style.setProperty(key, val + '');
-
-		setCSSVar('--x-landing-point', `${landingPoint}px`);
-
-		setCSSVar('--duration-chaos', `${durationChaos}ms`);
-
-		setCSSVar('--x1', `${x1}`);
-		setCSSVar('--x2', `${x2}`);
-		setCSSVar('--x3', `${x3}`);
-		setCSSVar('--x4', `${x4}`);
-
-		setCSSVar('--y1', `${y1}`);
-		setCSSVar('--y2', `${y2}`);
-		setCSSVar('--y3', `${y3}`);
-		setCSSVar('--y4', `${y4}`);
+		setCSSVar('--y4', `${round(max(mapRange(abs(degree - 180), 0, 180, force, -force), 0), 4)}`);
 
 		// set --width and --height here
 		setCSSVar(
 			'--width',
-			`${isCircle ? particleSize : Math.round(Math.random() * 4) + particleSize / 2}px`
+			`${isCircle ? particleSize : mathRound(random() * 4) + particleSize / 2}px`
 		);
 		setCSSVar(
 			'--height',
-			(isCircle ? particleSize : Math.round(Math.random() * 2) + particleSize) + 'px'
+			(isCircle ? particleSize : mathRound(random() * 2) + particleSize) + 'px'
 		);
 
-		setCSSVar('--rotation', `${rotationTransforms[rotationIndex].join()}`);
-		setCSSVar('--rotation-duration', `${rotation}ms`);
+		setCSSVar('--rotation', `${rotationTransform.toString(2).padStart(3, '0').split('')}`);
+		setCSSVar(
+			'--rotation-duration',
```

Comment:
Renaming the CSS variables would also improve the bundle size. Not sure if you care if they are readable or not... But shortening them all would probably reduce the bundle size to ~2.0kb min+gzip.